cmd/vulcan-snyk/main.go

```diff
@@ -0,0 +1,261 @@
+		if code != 0 {
+			return fmt.Errorf("%s", output)
+		}
+
+		logger.Infof("auth: %s", output)
+		if err != nil {
+			return err
+		}
+
+		output, _, _ = command.Execute(ctx, logger, "snyk", append([]string{"test", repoPath, "--all-sub-projects", "--json"})...)
+
+		r := SnykResponse{}
+		err = json.Unmarshal(output, &r)
+		if err != nil {
+			return err
+		}
+
+		// Group vulnerabilities by their Snyk Reference ID and module name, also filter out license issues
+		snykVulnerabilitiesMap := make(map[string]map[string][]SnykVulnerability)
+		for _, v := range r.Vulnerabilities {
+			if v.Type == "license" {
+				continue
+			}
+
+			_, ok := snykVulnerabilitiesMap[v.ID]
+			if !ok {
+				snykVulnerabilitiesMap[v.ID] = make(map[string][]SnykVulnerability)
+			}
+
+			snykVulnerabilitiesMap[v.ID][v.ModuleName] = append(snykVulnerabilitiesMap[v.ID][v.ModuleName], v)
+		}
+
+		vulns := []report.Vulnerability{}
+		// for each pair of (snyk vuln & module name), create a vulcan vuln
+		for _, snykModulesMap := range snykVulnerabilitiesMap {
+			for moduleName, snykIssues := range snykModulesMap {
+				vulcanVulnerability := &report.Vulnerability{}
+
+				vulcanVulnerability.Summary = snykIssues[0].Title + ": " + moduleName
+				vulcanVulnerability.Description = extractOverview([]byte(snykIssues[0].Description))
+				vulcanVulnerability.Details = extractDetails([]byte(snykIssues[0].Description))
+				vulcanVulnerability.ImpactDetails = createImpactDetails(snykIssues)
+				vulcanVulnerability.Score = snykIssues[0].CVSSScore
+				vulcanVulnerability.Recommendations = extractRecommendations([]byte(snykIssues[0].Description))
+
+				cweCount := len(snykIssues[0].Identifiers.CWE)
+				if cweCount > 0 {
+					if cweCount > 1 {
+						logger.Infof("Multiple CWE found (SNYK ID: %f). Storing the first CWE found.", snykIssues[0].CVSSScore)
+					}
+
+					cweID, err := strconv.Atoi(strings.ReplaceAll(snykIssues[0].Identifiers.CWE[0], "CWE-", ""))
+					if err != nil {
+						logger.Errorf("Not possible to convert %s to uint32", snykIssues[0].Identifiers.CWE[0])
+					} else {
+						vulcanVulnerability.CWEID = uint32(cweID)
+					}
+				}
+
+				for _, ref := range snykIssues[0].References {
+					vulcanVulnerability.References = append(vulcanVulnerability.References, ref.URL)
+				}
+
+				vulns = append(vulns, *vulcanVulnerability)
+			}
+		}
+
+		state.AddVulnerabilities(vulns...)
+
+		return nil
+	}
+	c := check.NewCheckFromHandler(checkName, run)
+
+	c.RunAndServe()
+}
+
+func createImpactDetails(vulnerabilities []SnykVulnerability) string {
+	res := ""
+	for _, vulnerability := range vulnerabilities {
+		str := "Introduced through: "
+		n := len(vulnerability.From)
+		for i := 0; i < n; i++ {
+			str = str + vulnerability.From[i]
+			if i < n-2 {
+				str = str + " > "
+			}
+		}
+		res = res + str + "\n"
+	}
+	return res
+}
+
+var regexpRemediationTagBegin = regexp.MustCompile(`(?i)<h2.*id="remediation".*</h2>`)
```

Comment:
I would actually avoid parsing HTML with regex and instead find the heading in the markdown and then only convert the relevant part to HTML. I think we can live with this, but I looked into it and `markdown.Parse` returns the AST and since the markdown is simple it should be pretty easy to find the relevant node.